.github/workflows/CI.yml

```diff
@@ -343,29 +344,76 @@ jobs:
       # Build and archive image into cache
       - uses: actions/checkout@v2
 
-      - name: Cache Image
-        id: cached_image
-        uses: actions/cache@v2
-        with:
-          key: "cached_image_${{ github.run_id }}"
-          path: "${{ env.PACKAGING_ROOT }}"
-
-      - name: restore cached binaries
+      - name: Restore cached binaries
         if: steps.cached_image.outputs.cache-hit != 'true'
         uses: actions/cache@v2
         with:
           key: "cached_binaries_${{ github.run_id }}"
           path: build
 
-      - name: Build Image
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to Public ECR
+        if: steps.cached_image.outputs.cache-hit != 'true'
+        id: login-ecr
+        uses: docker/login-action@v1
+        with:
+          registry: public.ecr.aws
+          username: ${{ secrets.INTEG_TEST_AWS_KEY_ID }}
+          password: ${{ secrets.INTEG_TEST_AWS_KEY_SECRET }}
+        env:
+          AWS_REGION: us-east-1
+
+      - name: Versioning for adot collector image
+        id: versioning
+        run: |
+          Version="`cat VERSION`-$GITHUB_RUN_ID"
+          echo "::set-output name=version::$Version"
+
+      #Build the adot collector image for two primary reasons:
+      #-Using the adot collector image to do the integration test
+      #-Export it for delivery version image in CD
+      #Documentation: https://github.com/docker/build-push-action
+      - name: Build adot collector image
+        id: docker_build
+        uses: docker/build-push-action@v2
         if: steps.cached_image.outputs.cache-hit != 'true'
-        run: docker build -t $IMAGE_NAME -f cmd/awscollector/Dockerfile .
+        with:
+          file: cmd/awscollector/Dockerfile
+          context: .
+          push: true
+          tags: public.ecr.aws/${{ env.ECR_REPO }}:${{ steps.versioning.outputs.version }}
+
+      #Export docker artifact since push and export cannot be in the same steps
+      #There were many different ways such as pulling the image and save as a tar file but
+      #we would like to get rid of load and also it would be better for us to build a second time
+      #since the image layers have been built before.
+      - name: Export ADOT Collector Image's artifact
```

Comment:
Why do we still need to export into an artifact? We should be able to pull directly anywhere that is currently fetching the artifact

Reply:
True thought. At first I thought we would need to export the artifcact in order to load the image for the CD. However, we can [pull directly from the public ECR and retag it. Fixed this.](https://github.com/aws-observability/aws-otel-collector/pull/860/files#diff-a64cbb6e953e697cb4302ab31af9ebbd95a8de6216f874bb0ffdefce8e3c1cd9R340-R343)